WcaOnRails/app/controllers/api/v0/competitions_controller.rb

```diff
@@ -8,4 +7,13 @@ def show
     end
     render json: competition
   end
+
+  def results
+  	competition = Competition.where(id: params[:id], showAtAll: true).firsts
+  	unless Competition
+  		render json: { error: "Competition with id #{params[:id]} not found" }, status: 404
+  		return
+  	end
+  	render json: Result.search_by_competition(competition.id, params: params)
+  end
```

Comment:
- `where(...).first` is effectively the same as `find_by(...)`
- I think that using `else` is cleaner than the `return`

I'd rewrite it a bit
```ruby
# competition.rb
# Add a handy scope to the Competition model
scope :visible, -> { where(showAtAll: true }
```
```ruby
# Here
def results
  competition = Competition.visible.find_by(id: params[:id])
  if competition
    render json: Result.search_by_competition(competition.id, params: params)
  else
    render status: :not_found, json: { error: "Competition with id #{params[:id]} not found." }
  end
end
```